src/sleep/sleep.c

```diff
@@ -302,7 +307,7 @@ static int execute_s2h(const SleepConfig *sleep_config) {
                 } else {
                         previous_discharge_rate = r;
                         assert(previous_discharge_rate != 0);
-                        suspend_interval = usec_sub_unsigned(last_capacity * USEC_PER_HOUR / previous_discharge_rate, 30 * USEC_PER_MINUTE);
+                        suspend_interval = usec_sub_unsigned(total_battery_capacity * USEC_PER_HOUR / previous_discharge_rate, 30 * USEC_PER_MINUTE);
```

Comment:
This is comparing the summed up capacities and rates, but the rate of one battery cannot be applied to the capacity of another, as they will likely be different? IE if one battery is has a rate of 50%/hour and another of 1%/hour, and they are both fully charged, the sleep time should be 101 hours, but this will calculate it as 3 hours. The intervals need to be computed separately.

Reply:
yes. will fix it.

Reply:
added now. Please review.